components/script/dom/document.rs

```diff
@@ -1438,16 +1427,16 @@ impl Document {
         }
 
         if cancel_state == EventDefault::Allowed {
-            let msg = EmbedderMsg::KeyEvent(ch, key, state, modifiers);
+            let msg = EmbedderMsg::Keyboard(keyboard_event.clone());
             self.send_to_embedder(msg);
 
             // This behavior is unspecced
             // We are supposed to dispatch synthetic click activation for Space and/or Return,
             // however *when* we do it is up to us.
             // Here, we're dispatching it after the key event so the script has a chance to cancel it
             // https://www.w3.org/Bugs/Public/show_bug.cgi?id=27337
-            match key {
-                Key::Space if state == KeyState::Released => {
+            match keyboard_event.key {
+                Key::Character(ref letter) if letter == " " && keyboard_event.state == KeyState::Up => {
```

Comment:
Due to changing from `Key::Space` to `Key::Character` with checking `" "`, I'm just curious, if the space is in `full width`, would it be handled in this matching arm? 🤔

Reply:
Ah, I see. Sorry for my misunderstanding and, yes, it will match for ASCII only so it should be fine to match the `" "` here like original way.

https://github.com/servo/servo/blob/511dfe8e1b062324fbfde0aac8d16cc5e20cf8ce/components/webdriver_server/keys.rs#L13-L15